### PR TITLE
refactor: move infverif task's condition script logic to cargo_make.rs

### LINKED ISSUE #172: Move infverif samples `condition_script` into `cargo_make.rs`
the logic of this condition script is complex enough that i think it'd be better to move it all into a function in `cargo_make.rs`. this also avoids needing to set `utils` to `pub` unnecessarily

_Originally posted by @wmmc88 in https://github.com/microsoft/windows-drivers-rs/pull/143#discussion_r1621445618_

### PR DESCRIPTION
**Pull request by svasista-ms**
Closes #172 

- Moved the InfVerif task's condition script logic from rust-driver-sample-makefile.toml to a function in cargo-make.rs.
- Ensured the utils module remains private.

### CODE REVIEW

File: crates/wdk-build/src/lib.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ pub mod cargo_make;
 pub mod metadata;
 /// Module for utility code related to the cargo-make experience for building
```
Comment (wmmc88) [inline crates/wdk-build/src/lib.rs]: If you're keeping this module doc comment, update it to reflact that its contents is meant to be internal and private (ie. not exported from crate)

**Review by NateD-MSFT — APPROVED** [review]

File: crates/wdk-build/src/lib.rs (hunk 1 above)
Comment (svasista-ms) [reply crates/wdk-build/src/lib.rs]: Moved the doc comment to utils module and removed it from lib.rs

File: crates/wdk-build/src/cargo_make.rs [hunk 2]
```diff
@@ -1052,6 +1062,38 @@ where
...
 }
 
+/// `cargo-make` condition script for `infverif` task in
+/// [`rust-driver-sample-makefile.toml`](../rust-driver-sample-makefile.toml)
+///
+/// # Errors
+///
+/// This function returns an error whenever it determines that the
+/// `infverif` `cargo-make` task should be skipped (i.e. when the WDK Version is
+/// bugged and does not contain /samples flag
```
Comment (wmmc88) [inline crates/wdk-build/src/cargo_make.rs]:
```suggestion
/// bugged and does not contain /samples flag)
```

File: crates/wdk-build/src/cargo_make.rs [hunk 3]
```diff
@@ -1052,6 +1062,38 @@ where
...
+///
+/// # Panics
+/// Panics if the `WDK_BUILD_DETECTED_VERSION` environment variable is not set.
+pub fn infverif_condition_script() -> anyhow::Result<()> {
```
Comment (wmmc88) [inline crates/wdk-build/src/cargo_make.rs]:
```suggestion
pub fn driver_sample_inverif_conidition_script() -> anyhow::Result<()> {
```

File: crates/wdk-build/rust-driver-sample-makefile.toml [hunk 4]
```diff
@@ -41,23 +41,5 @@ condition_script = '''
...
-		return Err::<(), String>(format!("Skipping InfVerif. InfVerif in WDK Build {wdk_build_number} is bugged and does not contain the /samples flag.").into());
-	}
-	Ok(())
-}).unwrap_or_else(|_|{
-	// panic contents would have already been printed to console
-	eprintln!("condition_script for infverif task panicked while executing. Defaulting to running inverif task.");
-	Ok(())
-})?
+wdk_build::cargo_make::infverif_condition_script()?
 '''
```
Comment (wmmc88) [inline crates/wdk-build/rust-driver-sample-makefile.toml]: Explicitly setting it to be a main returning `anyhow::Result<()>` ends up with more readable errors/output from rustscript. Without the explicit return type here, I think `rust-script` inserts some variant of `Box<dyn Error>` that ends up being a much messier output. I do the same thing for the `condition_script` for the `package-driver` task

```suggestion
fn main() -> anyhow::Result<()> {
    wdk_build::cargo_make::infverif_condition_script()
}
'''
```

File: crates/wdk-build/src/cargo_make.rs [hunk 5]
```diff
@@ -1052,6 +1062,38 @@ where
...
+/// [`rust-driver-sample-makefile.toml`](../rust-driver-sample-makefile.toml)
+///
+/// # Errors
+///
+/// This function returns an error whenever it determines that the
+/// `infverif` `cargo-make` task should be skipped (i.e. when the WDK Version is
+/// bugged and does not contain /samples flag
+///
+/// # Panics
+/// Panics if the `WDK_BUILD_DETECTED_VERSION` environment variable is not set.
```
Comment (wmmc88) [inline crates/wdk-build/src/cargo_make.rs]: This actually only panics if `CARGO_MAKE_CURRENT_TASK_NAME` is not set. See `condition_script` function's docs

File: crates/wdk-build/src/cargo_make.rs [hunk 6]
```diff
@@ -1052,6 +1062,38 @@ where
...
+pub fn infverif_condition_script() -> anyhow::Result<()> {
+    condition_script(|| {
+        let wdk_version = env::var(WDK_VERSION_ENV_VAR).expect(
+            "WDK_BUILD_DETECTED_VERSION should always be set by wdk-build-init cargo make task",
+        );
+        let wdk_build_number = str::parse::<u32>(&get_wdk_version_number(&wdk_version).unwrap())
```
Comment (wmmc88) [inline crates/wdk-build/src/cargo_make.rs]: `expect` instead of `unwrap`

**Review by wmmc88 — CHANGES_REQUESTED** [review]
Just some minor comments. Good to merge it after those are addressed

**Review by wmmc88 — APPROVED** [review]

**Review by hamzamust — APPROVED** [review]

**Review by ayodejiige — APPROVED** [review]

**Pull request merged**